src/Updater.php

```diff
@@ -498,7 +500,7 @@ protected function validatePhar($phar)
 
     protected function cleanupAfterError()
     {
-        //@unlink($this->getBackupPharFile());
+        @unlink($this->getBackupPharFile());
```

Comment:
That's one solution. The problem is that the errors were being thrown as exceptions as we use an error handler when checking PHAR signatures. If the handler is not restored, it would leak into subsequent code (in this case the unlink() calls with @ suppression ignored, but could also be the user's own code which is probably isn't desired behaviour)).